### DISCUSSION TITLE
Is Vercel suitable for complex architectures?

### DISCUSSION THREAD

**Discussion by d-asensio** [vercel/vercel#5181] [Help] [answered]
Hello all!

I am considering to deploy one project that I have been developing in my spare time to Vercel. After reading the documentation I still have to fill some gaps to see if it match my use case.

The project architecture consists of several services:

- **neo4j database**: This I assume that is not deployable to Vercel, and I have to look for some external service that allows me to deploy a docker container or whatever that can run this kind of DB (I know you are fully serverless now 👍 )

- **GraphQL server**: Easily deployable to Vercel as a serverless function, no doubts here.

- **Scrapy spider**: I use this service to scrap data from various sources, and I want to keep that data synchronized, so I need to run it periodically. It can run in a serverless function since it is written in python, but my doubt here is how to trigger it periodically, not involving an HTTP request. I found nothing about this topic on the docs, so I assume that is not doable.

- **Frontend**: As with the GraphQL no doubts here.

So basically, the whole thing works the following way.

1. The spider runs periodically and scraps various sources of data, such data is saved into a neo4j database.
2. GraphQL serves the data that is stored in neo4j and combines it with other sources, applies some business logic, etc..
3. The frontend makes requests to the GraphQL to request the data it needs.

I see the point of working 100% serverless, but in certain cases, it is unavoidable to run some things periodically or persist some data that is not completely static. In the case of persisting data, I understand that Vercel encourages you to use alternative services since one of the guides does exactly that (see [Create and Deploy a MongoDB-Powered Node.js API with Vercel](https://vercel.com/guides/deploying-a-mongodb-powered-api-with-node-and-vercel?query=Mongo#step-2:-creating-the-api)). I can deal with that, so the ability to trigger a serverless function periodically would cover my case.

The question also is, should I go with Vercel, or maybe it is better to consider Heroku or AWS? (maybe a mix?)

I know this can be very subjective but I would love to hear your opinion! 😊

Thank you in advance!

**Comment by d-asensio** [comment]
Hello? I would appreciate any comment :)

**Comment by styfle** [comment, marked as answer]
Hi @d-asensio 

Vercel it designed for websites, not large storage or long running tasks. Those should be hosted elsewhere so you can use the right service designed for those types of jobs.

1. neo4j database - will not work
2. GraphQL server - might work if you have a serverless package
3. Scrapy spider- will not work
4. Frontend - will work

Hope that helps!
